program_intent_engagement/apps/api/serializers.py

```diff
@@ -1,4 +1,30 @@
-# Serializers that can be shared across multiple versions of the API
-# should be created here. As the API evolves, serializers may become more
-# specific to a particular version of the API. In this case, the serializers
-# in question should be moved to versioned sub-package.
+"""
+Serializers for the program intent engagement API
+
+These serializers are shared across multiple API versions.
+Version-specific serializers are located in the versioned subpackage.
+"""
+from rest_framework import serializers
+
+from program_intent_engagement.apps.core.models import ProgramIntent, User
+
+
+class ProgramIntentSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the Program Intent Model
+    """
+
+    # todo: change to slug related field?
+    # todo: change user id to user?
+    user_id = serializers.IntegerField(required=True)
```

Comment:
question: Would it make sense to keep this as an IntegerField or would it be better to use `SlugRelatedField`? The actual post request json doesn't include the `user_id`, so I'm using the `lms_user_id` from the `request.user` to get the local `user_id `(see views.py). Then I'm adding this to the serializer. 

Michael had pointed me to `SlugRelatedField` which I think would have worked if I were getting the field from the model, but since I'm getting the user_id from the request itself I wasn't sure if it would be possible to incorporate that here using a `queryset`. Though I'm open to ideas because it would be nice if django can do some of the *~ magic ~* here as Michael had suggested.

Reply:
I know that Marcin has an `IntegerField` on his serializer as well: `user_id = serializers.IntegerField(source='user.lms_user_id')`. `source` is needed to pull the `lms_user_id` from the `User` object for serializing a model instance.

If we use the `ProgramIntentSerializer` to serialize a large number of ProgramIntents in bulk, though, we might run into a N+1 problem. I know that's way in the future, but it's just something to keep in mind.

Reply:
So I think this is now resolved because the model was changed such that user was replaced as `lms_user_id` and is no longer a FK.